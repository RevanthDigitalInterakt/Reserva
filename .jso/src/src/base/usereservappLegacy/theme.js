  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.theme = undefined;
  var tokens = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[0]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  var theme = exports.theme = {
    space: {
      quarck: 4,
      nano: 8,
      micro: 12,
      xxxs: 16,
      xxs: 24,
      xs: 32,
      sm: 40,
      md: 48,
      xl: 64,
      xxl: 80
    },
    shadows: {
      bottomBarShadow: '0px -3px 2px rgba(0,0,0,0.1)',
      topBarShadow: '0px 3px 2px rgba(0,0,0,0.1)'
    },
    colors: {
      preto: tokens.preto,
      vermelhoFechadoRSV: tokens.vermelhoFechadoRSV,
      marinho: tokens.marinho,
      neutroFrio2: tokens.neutroFrio2,
      neutroQuente2: tokens.neutroQuente2,
      vermelhoRSV: tokens.vermelhoRSV,
      verde: tokens.verde,
      neutroQuente1: tokens.neutroQuente1,
      offWhite: tokens.offWhite,
      offWhite15: tokens.offWhite15,
      whiteSecondary: tokens.whiteSecondary,
      neutroFrio1: tokens.neutroFrio1,
      leve: tokens.leve,
      vermelhoAlerta: tokens.vermelhoAlerta,
      verdeSucesso: tokens.verdeSucesso,
      amareloAtencao: tokens.amareloAtencao,
      dropDownBorderColor: tokens.dropDownBorderColor,
      white: tokens.white,
      fullBlack: tokens.fullBlack,
      divider: tokens.divider,
      backgoundInput: tokens.backgoundInput,
      backgroundApp: tokens.backgroundApp,
      transparente: 'transparent',
      modalBackDropColor: tokens.modalBackDropColor,
      borderButton: tokens.borderButton,
      dividerCounter: tokens.dividerCounter,
      backgoundDivider: tokens.backgoundDivider,
      progressTextColor: tokens.progressTextColor,
      backgroundMenuOpened: tokens.backgroundMenuOpened,
      searchBarTextColor: tokens.searchBarTextColor,
      borderColorProduct: tokens.borderColorProduct
    },
    fonts: {
      nunitoBlack: 'Nunito-Black',
      nunitoBlackItalic: 'Nunito-BlackItalic',
      nunitoBold: 'Nunito-Bold',
      nunitoBoldItalic: 'Nunito-BoldItalic',
      nunitoExtraBold: 'Nunito-ExtraBold',
      nunitoExtraBoldItalic: 'Nunito-ExtraBoldItalic',
      nunitoExtraLight: 'Nunito-ExtraLight',
      nunitoExtraLightItalic: 'Nunito-ExtraLightItalic',
      nunitoItalic: 'Nunito-Italic',
      nunitoLight: 'Nunito-Light',
      nunitoLightItalic: 'Nunito-LightItalic',
      nunitoRegular: 'Nunito-Regular',
      nunitoSemiBold: 'Nunito-SemiBold',
      nunitoSemiBoldItalic: 'Nunito-SemiBoldItalic',
      reservaDisplayRegular: 'ReservaDisplay-Regular',
      reservaSerifBlack: 'ReservaSerif-Black',
      reservaSerifBlackItalic: 'ReservaSerif-BlackItalic',
      reservaSerifBold: 'ReservaSerif-Bold',
      reservaSerifBoldItalic: 'ReservaSerif-BoldItalic',
      reservaSerifItalic: 'ReservaSerif-Italic',
      reservaSerifLight: 'ReservaSerif-Light',
      reservaSerifLightItalic: 'ReservaSerif-LightItalic',
      reservaSerifMedium: 'ReservaSerif-Medium',
      reservaSerifMediumItalic: 'ReservaSerif-MediumItalic',
      reservaSerifRegular: 'ReservaSerif-Regular',
      reservaSerifThin: 'ReservaSerif-Thin',
      reservaSerifThinItalic: 'ReservaSerif-ThinItalic',
      reservaSansBlack: 'ReservaSans-Black',
      reservaSansBlackItalic: 'ReservaSans-BlackItalic',
      reservaSansBold: 'ReservaSans-Bold',
      reservaSansBoldItalic: 'ReservaSans-BoldItalic',
      reservaSansItalic: 'ReservaSans-Italic',
      reservaSansLight: 'ReservaSans-Light',
      reservaSansLightItalic: 'ReservaSans-LightItalic',
      reservaSansMedium: 'ReservaSans-Medium',
      reservaSansMediumItalic: 'ReservaSans-MediumItalic',
      reservaSansRegular: 'ReservaSans-Regular',
      reservaSansThin: 'ReservaSans-Thin',
      reservaSansThinItalic: 'ReservaSans-ThinItalic'
    },
    fontWeights: {
      light: 300,
      normal: 400,
      medium: 500,
      SemiBold: 600,
      bold: 700,
      black: 900
    },
    fontSizes: [4, 8, 12, 14, 16, 18, 20, 24, 27, 32, 48],
    radii: {
      pico: 2,
      nano: 5,
      xxxs: 10,
      xs: 15,
      sm: 20,
      pill: 500,
      infinity: 999999
    },
    borderWidths: {
      none: tokens.borderWidthNone,
      hairline: tokens.borderWidthHairline,
      thin: tokens.borderWidthThin,
      thick: tokens.borderWidthThick,
      heavy: tokens.borderWidthHeavy,
      borderDiscountTag: tokens.borderDiscountTag,
      stepper: 3
    }
  };
