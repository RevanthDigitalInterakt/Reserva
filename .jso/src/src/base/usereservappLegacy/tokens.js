  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.whiteSecondary = exports.white = exports.vermelhoRSV = exports.vermelhoFechadoRSV = exports.vermelhoAlerta = exports.verdeSucesso = exports.verde = exports.searchBarTextColor = exports.progressTextColor = exports.preto = exports.offWhite15 = exports.offWhite = exports.neutroQuente2 = exports.neutroQuente1 = exports.neutroFrio2 = exports.neutroFrio1 = exports.modalBackDropColor = exports.marinho = exports.leve = exports.fullBlack = exports.dropDownBorderColor = exports.dividerCounter = exports.divider = exports.borderWidthThin = exports.borderWidthThick = exports.borderWidthNone = exports.borderWidthHeavy = exports.borderWidthHairline = exports.borderDiscountTag = exports.borderColorProduct = exports.borderButton = exports.backgroundMenuOpened = exports.backgroundApp = exports.backgoundInput = exports.backgoundDivider = exports.amareloAtencao = undefined;
  var white = exports.white = '#FFF';
  var fullBlack = exports.fullBlack = '#000';

  // cores primarias
  var preto = exports.preto = '#333333';
  var vermelhoFechadoRSV = exports.vermelhoFechadoRSV = '#5A0000';
  var marinho = exports.marinho = '#13294B';
  var neutroFrio2 = exports.neutroFrio2 = '#8A8C8E';
  var neutroQuente2 = exports.neutroQuente2 = '#B1ACA8';
  var divider = exports.divider = '#D8D9DA';
  var dividerCounter = exports.dividerCounter = '#EAEAEA';
  var progressTextColor = exports.progressTextColor = '#979797';
  var searchBarTextColor = exports.searchBarTextColor = '#666666';

  // cores secundarias
  var vermelhoRSV = exports.vermelhoRSV = '#A00000';
  var verde = exports.verde = '#22372B';
  var neutroQuente1 = exports.neutroQuente1 = '#D7CFCA';
  var offWhite = exports.offWhite = '#D6D1C4';
  var offWhite15 = exports.offWhite15 = '#F9F8F6';
  var whiteSecondary = exports.whiteSecondary = '#FAFAFA';
  var neutroFrio1 = exports.neutroFrio1 = '#D8D9DA';
  var leve = exports.leve = '#F9E1DD';

  // cores do sistema
  var vermelhoAlerta = exports.vermelhoAlerta = '#EF1E1E';
  var verdeSucesso = exports.verdeSucesso = '#38A238';
  var amareloAtencao = exports.amareloAtencao = '#FEA528';
  var dropDownBorderColor = exports.dropDownBorderColor = '#DDDDDD';

  // larguras de borda
  var borderWidthNone = exports.borderWidthNone = 0;
  var borderWidthHairline = exports.borderWidthHairline = 1;
  var borderWidthThin = exports.borderWidthThin = 2;
  var borderWidthThick = exports.borderWidthThick = 4;
  var borderWidthHeavy = exports.borderWidthHeavy = 8;
  var borderDiscountTag = exports.borderDiscountTag = 12;

  // cores de borda
  var borderButton = exports.borderButton = '#333333';
  var borderColorProduct = exports.borderColorProduct = '#757575';

  // backgrounds
  var backgroundMenuOpened = exports.backgroundMenuOpened = '#F1F1F1';
  var backgroundApp = exports.backgroundApp = '#F9F9F9';
  var modalBackDropColor = exports.modalBackDropColor = '#AFAFAF';
  var backgoundInput = exports.backgoundInput = '#F0F0F0';
  var backgoundDivider = exports.backgoundDivider = '#F5F5F5';
