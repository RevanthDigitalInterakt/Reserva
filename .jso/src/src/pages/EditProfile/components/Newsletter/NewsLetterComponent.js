  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _asyncToGenerator2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _slicedToArray2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[2]));
  var _react = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[3]));
  var _newsLetter = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[4]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  function NewsLetterComponent(_ref) {
    var userEmail = _ref.userEmail,
      value = _ref.value,
      handleToogleNewsLetterState = _ref.handleToogleNewsLetterState;
    var _useMutation = (0, _$$_REQUIRE(_dependencyMap[5]).useMutation)(_newsLetter.default),
      _useMutation2 = (0, _slicedToArray2.default)(_useMutation, 1),
      updateNewsLetter = _useMutation2[0];
    var handleUpdateNewsLetter = (0, _react.useCallback)(/*#__PURE__*/(0, _asyncToGenerator2.default)(function* () {
      var _yield$updateNewsLett = yield updateNewsLetter({
          variables: {
            email: userEmail,
            isNewsletterOptIn: !value
          }
        }),
        data = _yield$updateNewsLett.data;
      if (data.subscribeNewsletter) {
        handleToogleNewsLetterState(!value);
      }
    }), [userEmail, value]);
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[6]).jsx)(_$$_REQUIRE(_dependencyMap[7]).Box, {
      mb: "xs",
      flexDirection: "row",
      zIndex: 2,
      children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[6]).jsx)(_$$_REQUIRE(_dependencyMap[8]).Checkbox, {
        testID: "com.usereserva:id/newsletter_button",
        fontFamily: "nunitoRegular",
        fontSize: 12,
        color: "dropDownBorderColor",
        selectedColor: "preto",
        width: "100%",
        checked: value,
        onCheck: handleUpdateNewsLetter,
        optionName: "Desejo receber e-mails com promo\xE7\xF5es das marcas Reserva."
      })
    });
  }
  var _default = exports.default = NewsLetterComponent;
