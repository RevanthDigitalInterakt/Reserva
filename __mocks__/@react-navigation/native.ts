const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
