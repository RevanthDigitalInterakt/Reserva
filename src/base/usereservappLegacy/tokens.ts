export const white = '#FFF';
export const fullBlack = '#000';

// cores primarias
export const preto = '#333333';
export const vermelhoFechadoRSV = '#5A0000';
export const marinho = '#13294B';
export const neutroFrio2 = '#8A8C8E';
export const neutroQuente2 = '#B1ACA8';
export const divider = '#D8D9DA';
export const dividerCounter = '#EAEAEA';
export const progressTextColor = '#979797';
export const searchBarTextColor = '#666666';

// cores secundarias
export const vermelhoRSV = '#A00000';
export const verde = '#22372B';
export const neutroQuente1 = '#D7CFCA';
export const offWhite = '#D6D1C4';
export const offWhite15 = '#F9F8F6';
export const whiteSecondary = '#FAFAFA';
export const neutroFrio1 = '#D8D9DA';
export const leve = '#F9E1DD';

// cores do sistema
export const vermelhoAlerta = '#EF1E1E';
export const verdeSucesso = '#38A238';
export const amareloAtencao = '#FEA528';
export const dropDownBorderColor = '#DDDDDD';

// larguras de borda
export const borderWidthNone = 0;
export const borderWidthHairline = 1;
export const borderWidthThin = 2;
export const borderWidthThick = 4;
export const borderWidthHeavy = 8;
export const borderDiscountTag = 12;

// cores de borda
export const borderButton = '#333333';
export const borderColorProduct = '#757575';

// backgrounds
export const backgroundMenuOpened = '#F1F1F1';
export const backgroundApp = '#F9F9F9';
export const modalBackDropColor = '#AFAFAF';
export const backgoundInput = '#F0F0F0';
export const backgoundDivider = '#F5F5F5';
