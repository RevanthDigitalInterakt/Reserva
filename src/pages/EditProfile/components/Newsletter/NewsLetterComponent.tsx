import { Box, Checkbox } from '@usereservaapp/reserva-ui';
import React, { useCallback } from 'react';
import { useMutation } from '@apollo/client';
import subscribeNewsLetter from '../../../../graphql/profile/newsLetter';

interface INewsLetterProps {
  userEmail: string;
  value: boolean;

  handleToogleNewsLetterState: (value: boolean) => void
}
function NewsLetterComponent({
  userEmail,
  value,
  handleToogleNewsLetterState,
}: INewsLetterProps): JSX.Element {
  const [updateNewsLetter] = useMutation(subscribeNewsLetter);

  const handleUpdateNewsLetter = useCallback(async (): Promise<void> => {
    const { data } = await updateNewsLetter({
      variables: {
        email: userEmail,
        isNewsletterOptIn: !value,
      },
    });

    if (data.subscribeNewsletter) {
      handleToogleNewsLetterState(!value);
    }
  }, [userEmail, value]);

  return (
    <Box mb="xs" flexDirection="row" zIndex={2}>
      <Checkbox
        testID="com.usereserva:id/newsletter_button"
        fontFamily="nunitoRegular"
        fontSize={12}
        color="dropDownBorderColor"
        selectedColor="preto"
        width="100%"
        checked={value}
        onCheck={handleUpdateNewsLetter}
        optionName="Desejo receber e-mails com promoções das marcas Reserva."
      />
    </Box>
  );
}

export default NewsLetterComponent;
