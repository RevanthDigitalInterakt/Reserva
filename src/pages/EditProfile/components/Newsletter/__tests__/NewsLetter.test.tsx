import { ThemeProvider } from 'styled-components/native';
import React from 'react';
import {
  act, fireEvent, render, screen,
} from '@testing-library/react-native';
import { MockedProvider } from '@apollo/client/testing';
import NewsLetterComponent from '../NewsLetterComponent';
import subscribeNewsLetter from '../../../../../graphql/profile/newsLetter';
import { theme } from '../../../../../base/usereservappLegacy/theme';

const NewsLetterParams = {
  value: false,
  userEmail: 'user@gmail.com',
  handleToogleNewsLetterState: jest.fn(),
};

const apolloMocks = [
  {
    request: {
      query: subscribeNewsLetter,
      variables: {
        email: 'user@gmail.com',
        isNewsletterOptIn: true,
      },
    },
    result: {
      data: {
        subscribeNewsletter: true,
      },
    },
  },
];

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <MockedProvider mocks={apolloMocks} addTypename={false}>
      <NewsLetterComponent {...NewsLetterParams} />
    </MockedProvider>
  </ThemeProvider>
);

describe('NewsLetter Component', () => {
  it('renders without error and match snapshot', () => {
    render(TestingComponent);
    expect(screen.getByTestId('com.usereserva:id/newsletter_button')).toBeVisible();
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('Checks if the button is being clicked', async () => {
    render(TestingComponent);
    await act(async () => {
      await fireEvent.press(screen.getByTestId('com.usereserva:id/newsletter_button'));
    });

    await act(async () => {
    });

    // @TOTO - Implementar toHaveBeenCalledWith para verificar os parametros.
    expect(NewsLetterParams.handleToogleNewsLetterState).toHaveBeenCalledTimes(1);
  });
});
