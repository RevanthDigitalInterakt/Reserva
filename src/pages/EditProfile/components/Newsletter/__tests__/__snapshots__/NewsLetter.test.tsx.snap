// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsLetter Component renders without error and match snapshot 1`] = `
<View
  flexDirection="row"
  mb="xs"
  style={
    Array [
      Object {
        "flexDirection": "row",
        "marginBottom": 32,
        "zIndex": 2,
      },
    ]
  }
  zIndex={2}
>
  <View
    alignItems="center"
    flexDirection="row"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        },
      ]
    }
    width="100%"
  >
    <View
      alignItems="center"
      style={
        Array [
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="com.usereserva:id/newsletter_button"
      >
        <RNSVGSvgView
          bbHeight={15}
          bbWidth={15}
          color="dropDownBorderColor"
          focusable={false}
          size={15}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Array [
                Object {
                  "color": "#DDDDDD",
                  "height": 15,
                  "width": 15,
                },
              ],
              Object {
                "flex": 0,
                "height": 15,
                "width": 15,
              },
            ]
          }
        >
          <RNSVGGroup>
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="100%"
              bbWidth="100%"
              focusable={false}
              height="100%"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Array [
                    Object {
                      "height": "100%",
                      "width": "100%",
                    },
                  ],
                  Object {
                    "flex": 0,
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
              vbHeight={15}
              vbWidth={15}
              width="100%"
            >
              <RNSVGGroup>
                <RNSVGGroup
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                      "stroke",
                    ]
                  }
                  stroke={4292730333}
                >
                  <RNSVGRect
                    height={15}
                    propList={
                      Array [
                        "stroke",
                      ]
                    }
                    rx={2}
                    stroke={null}
                    width={15}
                    x={0}
                    y={0}
                  />
                  <RNSVGRect
                    fill={null}
                    height={14}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    rx={1.5}
                    width={14}
                    x={0.5}
                    y={0.5}
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      ml="nano"
      style={
        Array [
          Object {
            "marginLeft": 8,
          },
        ]
      }
    >
      <Text
        fontFamily="nunitoRegular"
        fontSize={12}
        style={
          Array [
            Object {
              "fontFamily": "Nunito-Regular",
              "fontSize": 12,
            },
          ]
        }
        variant="botaoFiltrarEOrdenarProdutos"
      >
        Desejo receber e-mails com promoções das marcas Reserva.
      </Text>
    </View>
  </View>
</View>
`;
